pkg/dependencies/dependents.go

```diff
@@ -0,0 +1,242 @@
+
+			if len(isDependency.DependencyPackage.Namespaces[0].Names[0].Versions) == 0 {
+				findMatchingDepPkgVersionIDs, err := FindDepPkgVersionIDs(ctx, gqlClient, isDependency.DependencyPackage.Type,
+					isDependency.DependencyPackage.Namespaces[0].Namespace,
+					isDependency.DependencyPackage.Namespaces[0].Names[0].Name, isDependency.VersionRange)
+				if err != nil {
+					return nil, fmt.Errorf("error from FindMatchingDepPkgVersionIDs:%w", err)
+				}
+				depPkgIds = append(depPkgIds, findMatchingDepPkgVersionIDs...)
+			} else {
+				depPkgIds = append(depPkgIds, isDependency.DependencyPackage.Namespaces[0].Names[0].Versions[0].Id)
+			}
+
+			for _, depPkgId := range depPkgIds {
+				// Skip "guac" files.
+				if isDependency.DependencyPackage.Type == "guac" && isDependency.DependencyPackage.Namespaces[0].Namespace == "files" {
+					continue
+				}
+
+				// Inside the loop where you iterate through dependencies
+				updatePackagesAndNames(idToName, packages, depPkgId, pkgId, depPkgName, pkgName, dependencyEdges, dependentEdges)
+
+				// Update the edges with pkgId and depPkgId.
+				dependentEdges[depPkgId] = append(dependentEdges[depPkgId], pkgId) // pkgId is dependent on depPkgId
+				dependencyEdges[pkgId] = append(dependencyEdges[pkgId], depPkgId)  // depPkgId is a dependency of pkgId
+			}
+		}
+	}
+
+	return packages, nil
+}
+
+// updatePackagesAndNames updates the mapping of package IDs to their names, and constructs the dependency graph.
+// It takes a set of parameters including maps for ID to name conversion, packages, dependency and dependent edges,
+// and information about the package and its dependency such as their IDs, names, types, and namespaces.
+// This function skips processing for "guac" files in the "files" namespace and updates the provided maps with
+// the relationships between packages and their dependencies. It leverages traverseGraph to find all packages
+// that are either dependencies of or dependents on the given package, and updates the packages map accordingly.
+func updatePackagesAndNames(idToName map[string]string, packages map[string]dependencyNode, depPkgId, pkgId, depPkgName, pkgName string, dependencyEdges, dependentEdges map[string][]string) {
+	// Map the IDs to their names.
+	idToName[depPkgId] = depPkgName
+	idToName[pkgId] = pkgName
+
+	// First, we need to find all the packages that are dependencies of pkgName.
+	// We need to add them all to the dependencies of all nodes that have pkgName as a dependent.
+	// Note that we are only searching for dependencies of pkgName from the edges that have scanned so far
+	dependencyPackages := traverseGraph(depPkgId, dependencyEdges)
+
+	// Next we want to find all the packages that have pkgName as a dependency.
+	// Note that we are only searching of packages with pkgName as a dependency from the edges that have scanned so far.
+	// This dependentPackages map finds all packages that have pkgName as a dependent out of our pre-scanned packages.
```

Comment:
On line 159, it should be `that have pkgName as a dependency` instead of `that have pkgName as a dependent`.

Reply:
Changed the wording, thanks!